src/BenchmarkDotNet/Configs/ImmutableConfigBuilder.cs

```diff
@@ -61,7 +61,7 @@ public static ImmutableConfig Create(IConfig source)
                 uniqueRules,
                 uniqueRunnableJobs,
                 source.UnionRule,
-                source.ArtifactsPath,
+                source.ArtifactsPath ?? DefaultConfig.Instance.ArtifactsPath,
```

Comment:
@wojtpl2 could you please add a test to `ImmutableConfigTests` that ensures that `ImmutableConfig.ArtifactsPath` is never null? I just want to make sure we don't regress in the future.